stream-listener/twitter_stream_listen.py

```diff
@@ -0,0 +1,125 @@
+"""
+Example program for using Twitter Stream API. Prints messages for particular user ids
+or for 'social' track as fast as possible. Use -h for help.
+
+From:
+https://github.com/ideoforms/python-twitter-examples/blob/master/twitter-stream-extract-links.py
+https://github.com/sixohsix/twitter/blob/master/twitter/stream_example.py
+"""
+
+import argparse
+import logging
+import sys
+
+import twitter
+import twitter.api
+from twitter.oauth import OAuth
+from twitter.stream import TwitterStream, Timeout, HeartbeatTimeout, Hangup
+from twitter.util import printNicely
+
+log = None
+
+DEFAULT_CONFIG_FILE = './config.py'
+TWITTER_API = None
+
+def ensure_api(auth=None):
+    """Return authenticated Twitter API instance, instantiating if necessary."""
+    global TWITTER_API
+    if TWITTER_API is None:
+        log.debug("Initializing API")
+        TWITTER_API = twitter.Twitter(auth=auth)
+        try:
+            TWITTER_API.account.verify_credentials(include_entities=False, skip_status=True)
+            log.debug("Twitter account verified")
+        except twitter.api.TwitterHTTPError as ex:
+            log.error("Twitter account verification failed for provided credentials: %s", ex)
+            sys.exit(1)
+    return TWITTER_API
+
+def get_auth(config):
+    return OAuth(
+        config["access_key"], config["access_secret"], config["consumer_key"], config["consumer_secret"])
+
+def handle_tweet(tweet):
+    """Print out type of tweet and, if text tweet, the contents."""
+    if tweet is None:
+        printNicely("-- None --")
+    elif tweet is Timeout:
+        printNicely("-- Timeout --")
+    elif tweet is HeartbeatTimeout:
+        printNicely("-- Heartbeat Timeout --")
+    elif tweet is Hangup:
+        printNicely("-- Hangup --")
+    elif tweet.get('text'):
+        printNicely(tweet['text'])
+    else:
+        printNicely("-- Some data: " + str(tweet))
+
+def handles_to_ids(handles):
```

Comment:
Ah. The 100 is an API-level limit. I didn't want to bother writing the code to batch the requests if there were more than 100 handles supplied. I changed it to raise if there are more than 100 handles provided.